### ISSUE TITLE
Conflict `ASYNC110`-noqa and false-positive `RUF100` 

### ISSUE THREAD

**Issue by Olegt0rr** [astral-sh/ruff#12591]
## Code example

```python
# async_draft.py

import asyncio


class Worker:
    def __init__(self):
        self.tasks = []


async def test_worker():
    worker = Worker()

    while worker.tasks:  # noqa: ASYNC110
        await asyncio.sleep(1)

```

### Got `ASYNC110` highlighted issue

<img width="939" alt="Screenshot 2024-07-31 at 11 41 33" src="https://github.com/user-attachments/assets/5dcd36d4-1fd0-45a0-9154-a3a1bb7f545d">

Code highlighted with ruff plugin
BUT command false positive passes (it should be failed since ASYNC rule group is active)

```
ruff check async_draft.py
All checks passed!
```

### Locally muted `ASYNC110` issue with `noqa`

It's don't relevant for test (for me)

<img width="930" alt="Screenshot 2024-07-31 at 11 41 52" src="https://github.com/user-attachments/assets/b81105e9-cf87-40e8-8fdb-3c3e805e00b0">

### Got `RUF100`  issue

<img width="673" alt="Screenshot 2024-07-31 at 11 42 22" src="https://github.com/user-attachments/assets/0731a554-f2a8-45dc-9b1f-37de7d36e09c">

## Additional info

ruff 0.5.5

### Config

```toml

[tool.ruff.lint]
select = ["ASYNC"]
ignore = []
fixable = ["ALL"]
unfixable = []
```

**Comment by charliermarsh** [comment]
I don't see `ASYNC110` triggering on that example, so gonna close.

**Comment by Olegt0rr** [comment]
@charliermarsh, but why it's don't triggering? :)

**Comment by charliermarsh** [comment]
Oh sorry, I misunderstood. You need to add `--preview` for that to trigger:

```
❯ cargo run check foo.py --select ASYNC -n --preview
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/ruff check foo.py --select ASYNC -n --preview`
foo.py:12:5: ASYNC110 Use `anyio.Event` instead of awaiting `anyio.sleep` in a `while` loop
   |
10 |       worker = Worker()
11 |
12 |       while worker.tasks:
   |  _____^
13 | |         await asyncio.sleep(1)
   | |______________________________^ ASYNC110
   |

Found 1 error.
```

The `ASYNC` rules only apply to `trio` on stable.

**Comment by Olegt0rr** [comment]
Oh, looks like ruff plugin uses `--preview` mode, I'll create an issue for that case
Thanks
